### PR TITLE
fix: schema plus button for a table is visible after template menu is opened

### LINKED ISSUE #30219: Plus icon disappears when the user clicks on the button to add a query template from the database schema table.
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBKzYzQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f28a2386210aeb79c802ae96a1d8062c2fc6c7bb/CleanShot%202024-01-10%20at%2013.22.00.mp4)

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
Earlier the template menu was opened on the query editor page for a particular table from schema, the plus button used to open the template menu was hidden. This PR fixes the plus button to be visible even when the template menu is opened. This PR also makes sure that this scenario doesn't come up in future with the help of the test case added.

#### PR fixes following issue(s)
Fixes #30219 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a test case to ensure the visibility and interaction of the schema plus button in the Action editor page.
- **Enhancements**
	- Improved interaction with schema table elements in the DataSources class.
	- Enhanced UI behavior by adjusting the context menu visibility and width based on entity activity status.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug30219_Spect.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+
+const dataSources = ObjectsRegistry.DataSources;
+
+describe(
+  "Bug 21734: On Action editor page on the right side pane schema plus button interaction",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug30219_Spect.ts:6]: Is this correct bug id?

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug30219_Spect.ts:6]: Sorry I will update

**Comment by sneha122** [comment]
@AmanAgarwal041 Changes look good to me!! Tested the PR as well, looks good to me!!

File: app/client/cypress/support/Pages/DataSources.ts [hunk 2]
```diff
@@ -1951,4 +1955,26 @@ export class DataSources {
       1000,
     );
   }
+
+  private isTemplateMenuTriggerForSchemaTableVisible(index: number = 0) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:1959]: nit: TS helper are Pascal case `IsTemplateMenuTriggerForSchemaTableVisible`

File: app/client/cypress/support/Pages/DataSources.ts [hunk 3]
```diff
@@ -1951,4 +1955,26 @@ export class DataSources {
...
+    this.agHelper
+      .GetElement(`${this._dsSchemaContainer} ${this._dsSchemaEntityItem}`)
+      .eq(index)
+      .find(this._entityTriggerElement)
+      .should("be.visible");
+  }
+
+  public clickTemplateMenuForSchemaTable(index: number = 0) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:1967]: nit: TS helper are Pascal case `ClickTemplateMenuForSchemaTable`

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress changes.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**